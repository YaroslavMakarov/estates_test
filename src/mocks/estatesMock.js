export const estatesMock = {
    estates: {
        estates: [
            {
                'id': 1,
                'title': 'Shevchenko estate',
                'price': 30000,
                'address': {
                    'street': 'Melnychna',
                    'number': 12
                },
                'preview_img': 'https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg',
                'seller': 'Chack Norris',
                'description': 'The best from Chuck',
                'detailed_photos': [
                    'https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg',
                    'https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg',
                    'https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg'
                ]
            },
            {
                "id": 2,
                "title": "Yanukovich estate",
                "price": 3000000,
                "address": {
                    "street": "Prisident",
                    "number": 4
                },
                "preview_img": "https://biz.liga.net/images/general/2018/11/15/thumbnail-tw-20181115111659-8474.jpg?v=1542279867",
                "seller": "Chack Norris",
                "description": "The best from Chuck",
                "detailed_photos": [
                    "https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg",
                    "https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg",
                    "https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg"
                    ]
             },
        ],
    },
};