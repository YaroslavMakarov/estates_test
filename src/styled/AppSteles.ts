import styled from 'styled-components';

export const AppWrapper = styled.div`
    padding-top: 50px; 
`;