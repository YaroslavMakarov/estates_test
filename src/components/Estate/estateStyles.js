import styled from "styled-components";

export const EstateContainer = styled.div`
    box-sizing: border-box;
    padding: 15px;
`;
export const Img = styled.img.attrs(props => ({
    src: props.src
}))`
    display: block;
    width: 100%;
    height: 300px;
    object-fit: cover;
    margin-bottom: 10px;
`;
export const EstateDescription = styled.div`
    font-size: 16px;
    margin-bottom: 5px;
`;