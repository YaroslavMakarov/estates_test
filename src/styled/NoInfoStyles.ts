import styled from 'styled-components';

export const NoInfoContent = styled.div`
    font-size: 24px;
    font-wight: bold;
`;