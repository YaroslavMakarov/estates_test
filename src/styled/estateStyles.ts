import styled from 'styled-components';

export const EstateContainer = styled.div`
    box-sizing: border-box;
    padding: 15px;
`;
export const Img = styled.img.attrs(props => ({
    src: props.src
}))`
    display: block;
    width: 100%;
    height: ${props => props.id ? "400px" : "300px"};;
    object-fit: cover;
    margin-bottom: 10px;
`;
export const EstateDescription = styled.div`
    font-size: ${props => props.id ? "22px" : "16px"};
    margin-bottom: ${props => props.id ? "10px" : "5px"};
`;