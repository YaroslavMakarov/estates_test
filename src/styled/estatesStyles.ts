import styled from "styled-components";

export const EstatesWrapper = styled.div`
    display: grid;
    grid-template-columns: repeat(4, 250px);
    grid-gap: 15px;
    width: fit-content;
`; 