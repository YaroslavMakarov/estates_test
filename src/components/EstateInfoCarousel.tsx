import React from 'react';
import { Row, Col,
         Carousel, CarouselItem } from 'react-bootstrap';
import { Img } from '../styled/estateStyles';   

type Props = {
    photos: Array<string>;
};

const EstateInfoCarousel: React.FC<Props> = ({ photos }) => (
    <Row>
        <Col sm={{span: 6, offset: 3}}>
            <Carousel>
                {photos.map(photo => (
                    <CarouselItem
                        key={photo}
                    >
                        <Img src={photo} />
                    </CarouselItem>
                ))}
            </Carousel>
        </Col>
    </Row>
);

export default EstateInfoCarousel;
