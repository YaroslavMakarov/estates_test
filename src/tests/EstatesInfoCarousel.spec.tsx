import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { CarouselItem } from 'react-bootstrap';

import EstateInfoCarousel from '../components/EstateInfoCarousel';
import { estatesMock } from '../mocks/estatesMock';
import { Img } from '../styled/estateStyles';

describe('test EstateInfoCarousel', () => {
    let wrapper: ShallowWrapper;
    const photos = estatesMock.estates.estates[0].detailed_photos;

    beforeEach(() => {
        wrapper = shallow(<EstateInfoCarousel photos={photos} />);
    });

    it('get snapShot EstateInfoCarousel', () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('get CarouselItem count', () => {
        const item = wrapper.find(CarouselItem);
        expect(item).toHaveLength(3);
    });

    it('check Img src attr', () => {
        const carouselImg = wrapper.find(Img);
                
        carouselImg.forEach((img, index) => {
            expect(img.prop('src')).toBe(`${photos[index]}`);
        });
    });

});