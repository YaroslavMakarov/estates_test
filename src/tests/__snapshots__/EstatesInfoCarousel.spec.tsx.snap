// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test EstateInfoCarousel get snapShot EstateInfoCarousel 1`] = `
"<Row noGutters={false}>
  <Col sm={{...}}>
    <Carousel slide={true} fade={false} controls={true} indicators={true} defaultActiveIndex={0} interval={5000} keyboard={true} pause=\\"hover\\" wrap={true} touch={true} prevIcon={{...}} prevLabel=\\"Previous\\" nextIcon={{...}} nextLabel=\\"Next\\">
      <CarouselItem>
        <styled.img src=\\"https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg\\" />
      </CarouselItem>
      <CarouselItem>
        <styled.img src=\\"https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg\\" />
      </CarouselItem>
      <CarouselItem>
        <styled.img src=\\"https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg\\" />
      </CarouselItem>
    </Carousel>
  </Col>
</Row>"
`;
